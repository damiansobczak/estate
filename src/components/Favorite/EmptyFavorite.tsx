import React from "react";

class EmptyFavorite extends React.Component {
    render() {
        return (
            <div className="favorite__empty columns">
                <div className="column">
                    <span className="icon-x-circle"></span>
                    <h2>Upss.</h2>
                    <p>Aktualnie nie masz Å¼adnych zapisanych ofert.</p>
                </div>
            </div>
        );
    }
}

export default EmptyFavorite;