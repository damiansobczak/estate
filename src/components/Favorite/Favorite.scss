@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.favorite {
    display: none;
    background-color: #fff;
    box-shadow: 0 -30px 100px rgba(0, 0, 0, 0.7);
    min-height: 500px;
    max-height: 600px;
    width: 100%;
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 1000;
    padding: 30px 0;

    &--open {
        display: block;
    }

    &__header {
        display: flex;
        align-items: center;
        justify-content: space-between;
    }

    &__button {
        display: inline-flex;
        align-items: center;
        font-weight: 600;
        padding: 12px 18px;
        border-radius: 30px;
        background-color: #f6f6f6;
        border: 0;
        outline: 0;
        font-size: 0.85rem;
        cursor: pointer;
        transition: all 0.1s ease-in;
        margin-top: 10px;
        
        &:hover {
            background-color: $color-accent;
            color: #fff;
        }

        span {
            margin-right: 5px;
        }
    }

    &__close {
        position: absolute;
        right: 0;
        top: -45px;
        background-color: $color-accent;
        height: 30px;
        width: 30px;
        border-radius: 50%;
        display: flex;
        align-items: center;
        justify-content: center;
        color: #fff;
        box-shadow: 0 10px 20px rgba(0, 0, 0, 0.1);
        transition: opacity 0.2s ease-in-out;

        &:hover {
            opacity: 0.9;
        }
    }
}