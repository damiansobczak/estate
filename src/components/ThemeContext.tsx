import React, { MouseEvent } from "react";

interface ThemeContextInteface {
    isOpen: boolean;
    toggle?: (e: MouseEvent) => void;
    toggleFavorites?: (e: MouseEvent, details: any) => void;
    favorites: any;
    isFavorite: (id: number) => boolean;
    deleteFavorites?: (e: MouseEvent) => void;
}

export const ThemeContext = React.createContext<ThemeContextInteface>({
    isOpen: false,
    favorites: null,
    isFavorite: () => false
});
