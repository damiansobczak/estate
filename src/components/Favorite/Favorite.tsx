import React, { MouseEvent } from "react";
import "./Favorite.scss";
import Offer from "../Offer/Offer";
import { ThemeContext } from "../ThemeContext";

class Favorite extends React.Component<any, any> {
    render() {
        let value = this.context;
        return (
            <ThemeContext.Consumer>
                {value => (
                    <div className={`favorite ${value.isOpen ? `favorite--open` : ``}`}>
                        <div className="container">
                            <button className="favorite__close" onClick={(e) => { value.toggle && value.toggle(e) }}>
                                <span className="icon-x"></span>
                            </button>
                            <div className="favorite__header">
                                <h4 className="favorite__title">Ulubione</h4>
                                <button className="favorite__button" onClick={(event) => value.deleteFavorites && value.deleteFavorites(event)}>
                                    <span className="icon-trash-2"></span>Usu≈Ñ zapisane
                                </button>
                            </div>
                            <div className="favorite__items columns">
                                {value.favorites.map((item: any) => <Offer id={item.id} image={item.image} price={item.price} category={item.category} date={item.date} title={item.title} tags={item.tags} />)}
                            </div>
                        </div>
                    </div>
                )}
            </ThemeContext.Consumer>
        );
    }
}

export default Favorite;
