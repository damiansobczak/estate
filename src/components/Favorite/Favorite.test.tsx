import React from "react";
import { shallow, render } from "enzyme";
import renderer from "react-test-renderer";
import Favorite from "./Favorite";
import EmptyFavorite from "./EmptyFavorite";
import { ThemeContext } from "../ThemeContext";
import { BrowserRouter } from "react-router-dom";

const context = {
    isOpen: false,
    isFavorite: () => false,
    favorites: [
        {
            "id": 42,
            "title": "Osiedle Babina, ul. Babina 20/3A",
            "image": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-1.jpg",
            "price": "125,500zł",
            "category": "Sprzedaż",
            "date": "24 Czer, 2019",
            "tags": ["Liczba pokoi 3", "76m2", "Liczba pięter 1"]
        }
    ]
}

const context2 = {
    isOpen: false,
    isFavorite: () => false,
    favorites: []
}

describe("<Favorite />", () => {
    it("renders component correctly when selected", () => {
        const tree = renderer.create(
            <ThemeContext.Provider value={context}>
                <BrowserRouter>
                    <Favorite />
                </BrowserRouter>
            </ThemeContext.Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders component correctly when empty", () => {
        const tree = renderer.create(
            <ThemeContext.Provider value={context2}>
                <BrowserRouter>
                    <Favorite />
                </BrowserRouter>
            </ThemeContext.Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("shows correctly favorite item in modal", () => {
        const wrapper = render(
            <ThemeContext.Provider value={context}>
                <BrowserRouter>
                    <Favorite />
                </BrowserRouter>
            </ThemeContext.Provider>
        );
        expect(wrapper.find(".offer__title").html()).toContain("Osiedle Babina, ul. Babina 20/3A");
        expect(wrapper.find(".offer__media > img").prop("src")).toContain("http://localhost:8888/estate/wp-content/uploads/2020/01/image-1.jpg");
    });

    it("shows information when empty favorites", () => {
        const wrapper = render(
            <ThemeContext.Provider value={context2}>
                <BrowserRouter>
                    <Favorite />
                </BrowserRouter>
            </ThemeContext.Provider>
        );
        expect(wrapper.find(".favorite__empty").html()).toContain("Aktualnie nie masz");
    });
});
