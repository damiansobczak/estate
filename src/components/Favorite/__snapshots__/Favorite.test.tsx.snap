// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorite /> renders component correctly when empty 1`] = `
<div
  className="favorite "
>
  <div
    className="container"
  >
    <button
      className="favorite__close"
      onClick={[Function]}
    >
      <span
        className="icon-x"
      />
    </button>
    <div
      className="favorite__header"
    >
      <h4
        className="favorite__title"
      >
        Ulubione
      </h4>
      <button
        className="favorite__button"
        onClick={[Function]}
      >
        <span
          className="icon-trash-2"
        />
        Usuń zapisane
      </button>
    </div>
    <div
      className="favorite__empty columns"
    >
      <div
        className="column"
      >
        <span
          className="icon-x-circle"
        />
        <h2>
          Upss.
        </h2>
        <p>
          Aktualnie nie masz żadnych zapisanych ofert.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Favorite /> renders component correctly when selected 1`] = `
<div
  className="favorite "
>
  <div
    className="container"
  >
    <button
      className="favorite__close"
      onClick={[Function]}
    >
      <span
        className="icon-x"
      />
    </button>
    <div
      className="favorite__header"
    >
      <h4
        className="favorite__title"
      >
        Ulubione
      </h4>
      <button
        className="favorite__button"
        onClick={[Function]}
      >
        <span
          className="icon-trash-2"
        />
        Usuń zapisane
      </button>
    </div>
    <div
      className="favorite__items columns"
    >
      <a
        className="column is-6-tablet is-4-desktop"
        href="/oferta/42"
        onClick={[Function]}
      >
        <div
          className="offer__favorite "
          onClick={[Function]}
        >
          <span
            className="icon-heart"
          />
        </div>
        <div
          className="offer"
        >
          <div
            className="offer__media"
          >
            <img
              alt=""
              src="http://localhost:8888/estate/wp-content/uploads/2020/01/image-1.jpg"
            />
            <div
              className="offer__price"
            >
              125,500zł
            </div>
          </div>
          <div
            className="offer__content"
          >
            <div
              className="offer__meta"
            >
              <div
                className="offer__category"
              >
                Sprzedaż
              </div>
              <div
                className="offer__date"
              >
                24 Czer, 2019
              </div>
            </div>
            <div
              className="offer__title"
            >
              Osiedle Babina, ul. Babina 20/3A
            </div>
            <div
              className="offer__sep"
            />
            <div
              className="offer__tags"
            >
              Liczba pokoi 3, 
              76m2, 
              Liczba pięter 1, 
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
